File: lib/models/warehouse.dart
class Warehouse {
  final int warehouseId;
  final String warehouseName;
  final String location;

  Warehouse({
    required this.warehouseId,
    required this.warehouseName,
    required this.location,
  });

  factory Warehouse.fromJson(Map<String, dynamic> json) => Warehouse(
        warehouseId: json['warehouseId'] ?? 0,
        warehouseName: json['warehouseName'] ?? '',
        location: json['location'] ?? '',
      );

  Map<String, dynamic> toJson() => {
        'warehouseId': warehouseId,
        'warehouseName': warehouseName,
        'location': location,
      };
}

File: Models/DataBase/Warehouse.cs
using System.ComponentModel.DataAnnotations;

namespace quan_ly_chuoi_cua_hang_tap_hoa_api.Models.DataBase
{
    public class Warehouse
    {
        [Key]
        public int WarehouseId { get; set; }
        public string WarehouseName { get; set; }
        public string? Location { get; set; }

        public virtual ICollection<Inventory> Inventories { get; set; }
    }
}
